model/prelude.sail

```diff
@@ -191,114 +248,112 @@ val EXTZ : forall 'n 'm, 'm >= 'n. (implicit('m), bits('n)) -> bits('m)
+function bool_to_bits x = if x then {
+  0b1
+} else {
+  0b0
+}
 
 val bit_to_bool : bit -> bool
 function bit_to_bool b = match b {
-  bitone  => true,
-  bitzero => false
+  bitone => true,
+  bitzero => false,
 }
 
-val to_bits : forall 'l, 'l >= 0.(atom('l), int) -> bits('l)
-function to_bits (l, n) = get_slice_int(l, n, 0)
+val to_bits : forall 'l, 'l >= 0. (int('l), int) -> bits('l)
+function to_bits(l, n) = get_slice_int(l, n, 0)
 
 infix 4 <_s
 infix 4 >=_s
 infix 4 <_u
 infix 4 >=_u
 infix 4 <=_u
 
-val operator <_s  : forall 'n, 'n > 0. (bits('n), bits('n)) -> bool
+val operator <_s : forall 'n, 'n > 0. (bits('n), bits('n)) -> bool
 val operator >=_s : forall 'n, 'n > 0. (bits('n), bits('n)) -> bool
-val operator <_u  : forall 'n. (bits('n), bits('n)) -> bool
+val operator <_u : forall 'n. (bits('n), bits('n)) -> bool
 val operator >=_u : forall 'n. (bits('n), bits('n)) -> bool
 val operator <=_u : forall 'n. (bits('n), bits('n)) -> bool
 
-function operator <_s  (x, y) = signed(x) < signed(y)
-function operator >=_s (x, y) = signed(x) >= signed(y)
-function operator <_u  (x, y) = unsigned(x) < unsigned(y)
-function operator >=_u (x, y) = unsigned(x) >= unsigned(y)
-function operator <=_u (x, y) = unsigned(x) <= unsigned(y)
+function operator <_s(x, y) = signed(x) < signed(y)
+function operator >=_s(x, y) = signed(x) >= signed(y)
+function operator <_u(x, y) = unsigned(x) < unsigned(y)
+function operator >=_u(x, y) = unsigned(x) >= unsigned(y)
+function operator <=_u(x, y) = unsigned(x) <= unsigned(y)
 
 infix 7 >>
 infix 7 <<
 
-val "shift_bits_right" : forall 'n 'm. (bits('n), bits('m)) -> bits('n)
-val "shift_bits_left"  : forall 'n 'm. (bits('n), bits('m)) -> bits('n)
+val shift_bits_right = pure { _: "shift_bits_right" } : forall 'n 'm. (bits('n), bits('m)) -> bits('n)
+val shift_bits_left = pure { _: "shift_bits_left" } : forall 'n 'm. (bits('n), bits('m)) -> bits('n)
 
-val "shiftl" : forall 'm 'n, 'n >= 0. (bits('m), atom('n)) -> bits('m)
-val "shiftr" : forall 'm 'n, 'n >= 0. (bits('m), atom('n)) -> bits('m)
+val shiftl = pure { _: "shiftl" } : forall 'm 'n, 'n >= 0. (bits('m), int('n)) -> bits('m)
+val shiftr = pure { _: "shiftr" } : forall 'm 'n, 'n >= 0. (bits('m), int('n)) -> bits('m)
 
 overload operator >> = {shift_bits_right, shiftr}
 overload operator << = {shift_bits_left, shiftl}
 
 /* Ideally these would be sail builtin */
+function shift_right_arith64(v : bits(64), shift : bits(6)) -> bits(64) = let v128 : bits(128) = EXTS(v) in
+  (v128 >> shift)[
+    63..0
+  ]
 
-function shift_right_arith64 (v : bits(64), shift : bits(6)) -> bits(64) =
-    let v128 : bits(128) = EXTS(v) in
-    (v128 >> shift)[63..0]
-
-function shift_right_arith32 (v : bits(32), shift : bits(5)) -> bits(32) =
-    let v64 : bits(64) = EXTS(v) in
-    (v64 >> shift)[31..0]
+function shift_right_arith32(v : bits(32), shift : bits(5)) -> bits(32) = let v64 : bits(64) = EXTS(v) in
+  (v64 >> shift)[
+    31..0
+  ]
```

Comment:
This is really quite strange

Reply:
I haven't put much work currently into handling OCaml-style `let ... in ...`. I think if you had:
```
function shift_right_arith32(v : bits(32), shift : bits(5)) -> bits(32) = {
  let v64 : bits(64) = EXTS(v);
  (v64 >> shift)[31..0]
}
```
it would format it as expected. I was considering a pass to reformat in this style, but it'd be a bit more work.